CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,432 @@
+          }
+          """)
+
+        try! InitializerDeclSyntax("\(raw: generateInitializerDecl(children: node.children, optionalBaseAsMissing: false, currentIndentation: "  "))") {
+          let parameters = node.children.map { $0.swiftName }.joined(separator: ", ")
+
+          VariableDeclSyntax(
+            leadingTrivia: """
+              // Extend the lifetime of all parameters so their arenas don't get destroyed
+              // before they can be added as children of the new arena.
+
+              """,
+            letOrVarKeyword: .keyword(.let)) {
+              let withExtendedLifetimeParameters = TupleExprElementListSyntax {
+                TupleExprElementSyntax(expression: ExprSyntax("(SyntaxArena(), (\(raw: parameters)))"))
+              }
+              let closureSignature = ClosureSignatureSyntax(
+                input: ClosureSignatureSyntax.Input.input(
+                  ParameterClauseSyntax {
+                    FunctionParameterSyntax(firstName: .identifier("arena"))
+                    FunctionParameterSyntax(firstName: .wildcardToken())
+                  }
+                )
+              )
+              let layoutList = ArrayExprSyntax {
+                for child in node.children {
+                  if child.isOptional {
+                    ArrayElementSyntax(expression: ExprSyntax("\(raw: child.swiftName)?.raw"))
+                  } else {
+                    ArrayElementSyntax(expression: ExprSyntax("\(raw: child.swiftName).raw"))
+                  }
+                }
+              }
+              let layoutVariable = DeclSyntax("let layout: [RawSyntax?] = \(layoutList)")
+              let rawVariable = node.children.isEmpty ? DeclSyntax("let raw = RawSyntax.makeEmptyLayout(kind: SyntaxKind.\(raw: node.swiftSyntaxKind), arena: arena)")
+              : DeclSyntax("""
+                  let raw = RawSyntax.makeLayout(
+                    kind: SyntaxKind.\(raw: node.swiftSyntaxKind), from: layout, arena: arena,
+                    leadingTrivia: leadingTrivia, trailingTrivia: trailingTrivia)
+                  """)
+
+              let initializer: ExprSyntax = ExprSyntax("""
+                                                withExtendedLifetime(\(withExtendedLifetimeParameters)) { \(closureSignature)
+                                                  \(layoutVariable)
+                                                  \(rawVariable)
+                                                  return SyntaxData.forRoot(raw)
+                                                }
+                                                """)
+
+              PatternBindingSyntax(
+                pattern: PatternSyntax("data"),
+                typeAnnotation: TypeAnnotationSyntax(type: TypeSyntax("SyntaxData")),
+                initializer: InitializerClauseSyntax(
+                  value: initializer
+                )
+              )
+            }
+
+          ExprSyntax("self.init(data)")
+        }
```

Comment:
I couldn’t resist the temptation to play around with this to get rid of `layout` where it’s not needed. This is what I came up with. WDYT?

```swift
try! InitializerDeclSyntax("\(raw: generateInitializerDecl(children: node.children, optionalBaseAsMissing: false, currentIndentation: "  "))") {
  let parameters = node.children.map { $0.swiftName }.joined(separator: ", ")

  let closureSignature = ClosureSignatureSyntax(
    input: .input(
      ParameterClauseSyntax {
        FunctionParameterSyntax(firstName: .identifier("arena"))
        FunctionParameterSyntax(firstName: .wildcardToken())
      }
    )
  )
  let layoutList = ArrayExprSyntax {
    for child in node.children {
      ArrayElementSyntax(expression: MemberAccessExprSyntax(
        base: child.type.optionalChained(expr: ExprSyntax("\(raw: child.swiftName)")),
        dot: .periodToken(),
        name: "raw"
      ))
    }
  }

  let initializer = FunctionCallExprSyntax(
    calledExpression: ExprSyntax("withExtendedLifetime"),
    leftParen: .leftParenToken(),
    argumentList: TupleExprElementListSyntax {
      TupleExprElementSyntax(expression: ExprSyntax("(SyntaxArena(), (\(raw: parameters)))"))
    },
    rightParen: .rightParenToken(),
    trailingClosure: ClosureExprSyntax(signature: closureSignature) {
      if node.children.isEmpty {
        DeclSyntax("let raw = RawSyntax.makeEmptyLayout(kind: SyntaxKind.\(raw: node.swiftSyntaxKind), arena: arena)")
      } else {
        DeclSyntax("let layout: [RawSyntax?] = \(layoutList)")
        DeclSyntax("""
            let raw = RawSyntax.makeLayout(
              kind: SyntaxKind.\(raw: node.swiftSyntaxKind), from: layout, arena: arena,
              leadingTrivia: leadingTrivia, trailingTrivia: trailingTrivia)
            """)
      }
    StmtSyntax("return SyntaxData.forRoot(raw)")
  })

  VariableDeclSyntax(
    leadingTrivia: """
      // Extend the lifetime of all parameters so their arenas don't get destroyed
      // before they can be added as children of the new arena.

      """,
    .let,
    name: PatternSyntax("data"),
    type: TypeAnnotationSyntax(type: TypeSyntax("SyntaxData")),
    initializer: InitializerClauseSyntax(value: initializer)
  )

  ExprSyntax("self.init(data)")
}
```

Reply:
Copied and pasted!